HPCC-24069 Fix changeToWrite transaction bug

jakesmith opened the pull request:
Do not hold the SDS transaction mutex (dataRWLock) whilst trying
to establish a write lock in the RTM_DELETE_ON_DISCONNECT logic.
    
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: 82e0134695134a99855c0a7583633e15d5207dc4
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-188.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Did not build yet.
Commit ID: 82e0134695134a99855c0a7583633e15d5207dc4
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-188.ca-central-1.compute.internal
Sha: 82e0134695134a99855c0a7583633e15d5207dc4
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |874 |874 |0 |309 sec  (00:05:09) |
|test (thor) |812 |812 |0 |753 sec  (00:12:33) |
|test (roxie) |953 |953 |0 |412 sec  (00:06:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  644 sec (00:10:44) |  117 sec (00:01:57) |   23 sec (00:00:23) |   19 sec (00:00:19) | 1802 sec (00:30:02) |   18 sec (00:00:18) | 2632 sec (00:43:52) |

jakesmith commented:
@ghalliday - please review

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 82e0134695134a99855c0a7583633e15d5207dc4
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |46 sec  (00:00:46) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |874 |874 |0 |571 sec  (00:09:31) |
|test (thor) |812 |812 |0 |1028 sec  (00:17:08) |
|test (roxie) |953 |953 |0 |712 sec  (00:11:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  923 sec (00:15:23) |  123 sec (00:02:03) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2646 sec (00:44:06) |   17 sec (00:00:17) | 3756 sec (01:02:36) |

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: New commit.
Commit ID: d2a23e16ac3d4c5376be0fcd9b72d37f1f64ab79
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: d2a23e16ac3d4c5376be0fcd9b72d37f1f64ab79
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-33.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-33.ca-central-1.compute.internal
Sha: d2a23e16ac3d4c5376be0fcd9b72d37f1f64ab79
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |874 |874 |0 |305 sec  (00:05:05) |
|test (thor) |812 |812 |0 |745 sec  (00:12:25) |
|test (roxie) |953 |953 |0 |392 sec  (00:06:32) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  640 sec (00:10:40) |  117 sec (00:01:57) |   22 sec (00:00:22) |   18 sec (00:00:18) | 1769 sec (00:29:29) |   19 sec (00:00:19) | 2594 sec (00:43:14) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: d2a23e16ac3d4c5376be0fcd9b72d37f1f64ab79
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |27 sec  (00:00:27) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |572 sec  (00:09:32) |
|test (thor) |812 |812 |0 |999 sec  (00:16:39) |
|test (roxie) |953 |953 |0 |709 sec  (00:11:49) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  923 sec (00:15:23) |  122 sec (00:02:02) |   24 sec (00:00:24) |   17 sec (00:00:17) | 2633 sec (00:43:53) |   17 sec (00:00:17) | 3745 sec (01:02:25) |

richardkchapman commented on dali/base/dasds.cpp:
```diff
@@ -267,7 +272,12 @@ class CLCLockBlock : public CInterface
-        lock.changeToWrite(timeout);
+        /* NB: this intentionally clears the lock, before re-establishing a write lock
+        * because we do not want to keep this read lock whilst waiting for a write,
+        * as other write transactions would pile up.
```
And we would deadlock...

richardkchapman commented on dali/base/dasds.cpp:
```diff
@@ -7848,43 +7858,63 @@ void CCovenSDSManager::disconnect(ConnectionId id, bool deleteRoot, CLCLockBlock
-        if (lockBlock)
+        if (lockBlock) // NB: only passed if might need read->write
+        {
             lockBlock->changeToWrite(readWriteTimeout);
```
This makes me nervous!

ghalliday commented:
@richardkchapman it looks correct as far as can see.

jakesmith commented:
@ghalliday @richardkchapman - I've spotted an issue, don't merge yet.

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: New commit.
Commit ID: 0098e9617626aa01891cced284826a37c4bfbd6e
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: 0098e9617626aa01891cced284826a37c4bfbd6e
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-20.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-20.ca-central-1.compute.internal
Sha: 0098e9617626aa01891cced284826a37c4bfbd6e
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |51 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |31 sec  (00:00:31) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |23 sec  (00:00:23) |
|test (hthor) |874 |874 |0 |307 sec  (00:05:07) |
|test (thor) |812 |812 |0 |746 sec  (00:12:26) |
|test (roxie) |953 |953 |0 |412 sec  (00:06:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  637 sec (00:10:37) |  119 sec (00:01:59) |   23 sec (00:00:23) |   21 sec (00:00:21) | 1797 sec (00:29:57) |   18 sec (00:00:18) | 2625 sec (00:43:45) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 0098e9617626aa01891cced284826a37c4bfbd6e
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |46 sec  (00:00:46) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |575 sec  (00:09:35) |
|test (thor) |812 |812 |0 |1018 sec  (00:16:58) |
|test (roxie) |953 |953 |0 |704 sec  (00:11:44) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  921 sec (00:15:21) |  123 sec (00:02:03) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2627 sec (00:43:47) |   18 sec (00:00:18) | 3736 sec (01:02:16) |

jakesmith commented:
@ghalliday - please re-review

richardkchapman commented on dali/base/dasds.cpp:
```diff
@@ -270,18 +262,28 @@ class CLCLockBlock : public CInterface
-        lock.unlock();
-        lock.lockWrite(timeout);
-    }
+    void changeToWrite(unsigned timeout);
```
Should this have same prototype as above?

jakesmith replied:
yes, it's in conditional code which isn't on unless a #define is changed, but should be changed.

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: f365e7c14a560efe00a0f960b12525425223420a
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-18.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: New commit.
Commit ID: f365e7c14a560efe00a0f960b12525425223420a
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-18.ca-central-1.compute.internal
Sha: f365e7c14a560efe00a0f960b12525425223420a
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |17 sec  (00:00:17) |
|test (hthor) |874 |874 |0 |304 sec  (00:05:04) |
|test (thor) |812 |812 |0 |738 sec  (00:12:18) |
|test (roxie) |953 |953 |0 |405 sec  (00:06:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  640 sec (00:10:40) |  118 sec (00:01:58) |   22 sec (00:00:22) |   17 sec (00:00:17) | 1772 sec (00:29:32) |   18 sec (00:00:18) | 2596 sec (00:43:16) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: f365e7c14a560efe00a0f960b12525425223420a
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |55 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |46 sec  (00:00:46) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |874 |874 |0 |576 sec  (00:09:36) |
|test (thor) |812 |812 |0 |1017 sec  (00:16:57) |
|test (roxie) |953 |953 |0 |710 sec  (00:11:50) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  931 sec (00:15:31) |  122 sec (00:02:02) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2662 sec (00:44:22) |   18 sec (00:00:18) | 3781 sec (01:03:01) |

jakesmith committed:
HPCC-24069 Fix changeToWrite transaction bug

Do not hold the SDS transaction mutex (dataRWLock) whilst trying
to establish a write lock in the RTM_DELETE_ON_DISCONNECT logic.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

jakesmith commented:
@richardkchapman @ghalliday - left as a few commits for now, please review.

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: 3a9e9a0bfba9a8483c96e53d734c5fc905fbdc49
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-244.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: New commit.
Commit ID: 3a9e9a0bfba9a8483c96e53d734c5fc905fbdc49
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-244.ca-central-1.compute.internal
Sha: 3a9e9a0bfba9a8483c96e53d734c5fc905fbdc49
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |874 |874 |0 |312 sec  (00:05:12) |
|test (thor) |812 |812 |0 |755 sec  (00:12:35) |
|test (roxie) |953 |953 |0 |379 sec  (00:06:19) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  644 sec (00:10:44) |  118 sec (00:01:58) |   22 sec (00:00:22) |   18 sec (00:00:18) | 1775 sec (00:29:35) |   18 sec (00:00:18) | 2604 sec (00:43:24) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 3a9e9a0bfba9a8483c96e53d734c5fc905fbdc49
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |874 |874 |0 |565 sec  (00:09:25) |
|test (thor) |812 |812 |0 |1017 sec  (00:16:57) |
|test (roxie) |953 |953 |0 |720 sec  (00:12:00) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  937 sec (00:15:37) |  123 sec (00:02:03) |   26 sec (00:00:26) |   17 sec (00:00:17) | 2638 sec (00:43:58) |   18 sec (00:00:18) | 3770 sec (01:02:50) |

ghalliday commented:
@richardkchapman I haven't seen any issues.

jakesmith commented:
@richardkchapman - squashed.

HPCCSmoketest commented:
Process of PR-13765, label: hpcc-24069-sds-changeToWrite-bug starts now.
The reason of this test is: Forced to re-test.
Commit ID: 3819b3d43e051582f647d7c554134b5b0afb9332
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-227.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-227.ca-central-1.compute.internal
Sha: 3819b3d43e051582f647d7c554134b5b0afb9332
Build: success
Milestone:Install hpccsystems-platform-community_7.8.15-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |27 sec  (00:00:27) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |25 sec  (00:00:25) |
|test (hthor) |874 |874 |0 |311 sec  (00:05:11) |
|test (thor) |812 |812 |0 |749 sec  (00:12:29) |
|test (roxie) |953 |953 |0 |401 sec  (00:06:41) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  640 sec (00:10:40) |  121 sec (00:02:01) |   22 sec (00:00:22) |   19 sec (00:00:19) | 1792 sec (00:29:52) |   19 sec (00:00:19) | 2622 sec (00:43:42) |
